src/solidity-frontend/solidity_convert.cpp

```diff
@@ -0,0 +1,1898 @@
+        sub_type)) // TODO: Fix me! Inconsistency. Better to use Solidity compiler library?
+      return true;
+
+    if(
+      sub_type.is_struct() ||
+      sub_type.is_union()) // for "assert(sum > 100)", false || false
+      assert(!"struct or union is NOT supported");
+
+    new_type = gen_pointer_type(sub_type);
+    break;
+  }
+  case SolidityGrammar::TypeNameT::ArrayTypeName:
+  {
+    // Deal with array with constant size, e.g., int a[2]; Similar to clang::Type::ConstantArray
+    // array's typeDescription is in a compact form, e.g.:
+    //    "typeIdentifier": "t_array$_t_uint8_$2_storage_ptr",
+    //    "typeString": "uint8[2]"
+    // We need to extract the elementary type of array from the information provided above
+    // We want to make it like ["baseType"]["typeDescriptions"]
+    nlohmann::json array_elementary_type =
+      make_array_elementary_type(type_name);
+    typet the_type;
+    if(get_type_description(array_elementary_type, the_type))
+      return true;
+
+    assert(the_type.is_unsignedbv()); // assuming array size is unsigned bv
+    std::string the_size = get_array_size(type_name);
+    unsigned z_ext_value = std::stoul(the_size, nullptr);
+    new_type = array_typet(
+      the_type,
+      constant_exprt(
+        integer2binary(z_ext_value, bv_width(int_type())),
+        integer2string(z_ext_value),
+        int_type()));
+
+    break;
+  }
+  default:
+  {
+    msg.status(fmt::format(
+      "	@@@ got type name=SolidityGrammar::TypeNameT::{}",
+      SolidityGrammar::type_name_to_str(type)));
+    assert(!"Unimplemented type in rule type-name");
+    return true;
+  }
+  }
+
+  // TODO: More var decl attributes checks:
+  //    - Constant
+  //    - Volatile
+  //    - isRestrict
+
+  return false;
+}
+
+bool solidity_convertert::get_func_decl_ref_type(
+  const nlohmann::json &decl,
+  typet &new_type)
+{
+  // For FunctionToPointer decay:
+  // Get type when we make a function call:
+  //  - FunnctionNoProto: x = nondet()
+  //  - FunctionProto:    z = add(x, y)
+  // Similar to the function get_type_description()
+  SolidityGrammar::FunctionDeclRefT type =
+    SolidityGrammar::get_func_decl_ref_t(decl);
+
+  switch(type)
+  {
+  case SolidityGrammar::FunctionDeclRefT::FunctionNoProto:
+  {
+    code_typet type;
+
+    // Return type
+    const nlohmann::json &rtn_type = decl["returnParameters"];
+
+    typet return_type;
+    if(get_type_description(rtn_type, return_type))
+      return true;
+
+    type.return_type() = return_type;
+
+    if(!type.arguments().size())
+      type.make_ellipsis();
+
+    new_type = type;
+    break;
+  }
+  default:
+  {
+    msg.status(fmt::format(
+      "	@@@ Got type={}", SolidityGrammar::func_decl_ref_to_str(type)));
+    assert(!"Unimplemented type in auxiliary type to convert function call");
+    return true;
+  }
+  }
+
+  // TODO: More var decl attributes checks:
+  //    - Constant
+  //    - Volatile
+  //    - isRestrict
+  return false;
```

Comment:
remember to open an issue

Reply:
ACK.
will do it when extending the new frontend to support more benchmarks.